specs/src/shrex/shrex-sub.md

```diff
@@ -0,0 +1,200 @@
+- Serialization overhead SHOULD be minimal
+- Each EDS at a specific height SHOULD produce only a single notification
+
+## Protocol Components
+
+### PubSub Component
+
+The core publish-subscribe functionality for message distribution:
+
+**Interface Methods:**
+
+```text
+// Publish publishes data hash and height to the topic
+Publish(dataHash []byte, height uint64) -> error
+
+// Subscribe creates a new subscription to the topic
+Subscribe() -> (*Subscription, error)
+```
+
+**Requirements:**
+
+- Publishers MUST use the topic ID `/eds-sub/0.0.1`
+- The Publish method MUST validate input parameters before publishing
+- The Subscribe method MUST return a valid subscription or an error
+
+### Subscription Component
+
+The subscription component handles the receiving and processing of ShrEx/Sub notifications:
+
+**Interface Methods:**
+
+```text
+// Next blocks the caller until any new EDS DataHash notification arrives.
+// Returns only notifications which successfully pass validation.
+Next() -> (Notification, error) 
+```
+
+**Requirements:**
+
+- Implementations MUST maintain an active subscription to `/eds-sub/0.0.1` topic
+- Incoming notifications MUST be processed through the validation pipeline
+- Only validated notifications SHALL be distributed to registered listeners
+
+## Message Verification
+
+### Validation Interface
+
+**MessageValidator Interface:**
+
+```text
+// Validate validates a message from a peer
+Validate(peerID PeerID, message []byte) -> ValidationResult
+```
+
+**ValidationResult Values:**
+
+- `ACCEPT`: Message is valid and MUST be processed
+- `REJECT`: Message is invalid and MUST be discarded
+- `IGNORE`: Message is valid but duplicate/stale and SHOULD be ignored
+
+The validation interface is exposed specifically for integration with the Peer Manager, which implements this interface to validate incoming ShrEx/Sub notifications. The Peer Manager uses validation results to add the announcing peer to the appropriate data hash pool, making them available for future data requests
+
+### Validation Pipeline
+
+ShrEx/Sub implementations MUST implement the following validation process:
+
+**Sanity Checks:**
+
+- Hash length MUST be exactly 32 bytes
+- Hash MUST NOT be all zeros
+- Message size MUST NOT exceed protocol limits
+- Height MUST be a non-zero positive integer
+
+**Verification Checks:**
+
+- Height MUST correspond to a block header that is fully verified by the implementation
+- The block header at the given height MUST be included as part of the implementation's subjective chain
+- The data hash MUST match the data root hash from the verified header at the specified height
+
+### Component Interaction Flow
+
+```text
+1. PubSub receives message from network
+2. Subscription processes message through validation
+3. Validated messages sent to registered listeners
+```
+
+## Protocol Behavior
+
+### Publisher Behavior (BN)
+
+- **Trigger**: Publishers MUST publish when a new EDS becomes available locally
```

Comment:
please distinguish "new EDS becomes available locally" from a syncing BN

publishers should only publish message if subjectively it understands that it is synced up and that this EDS is from a height at the tip of the network